### PR TITLE
Speed up `make shellcheck` and add podman support

### PR DESCRIPTION
**Pull request by legoktm**
## Status

Ready for review

## Description of Changes

Instead of spawning a new container/shellcheck process for each file we
want to lint, we can pass the entire list of files to it and let it run
over all of them. On my laptop this sped up `make shellcheck` from 26
seconds to 1.5 seconds!

podman support is added using the same environment variable as #6216,
`USE_PODMAN=1`.

## Testing

* Insert some buggy code in a shell script and run `make shellcheck` and confirm it still reports an error and exits with a non-zero status code
* Install `shellcheck` from your package manager and make sure it works in native mode. (this will likely spit out a bunch of errors because of the version mismatch between your package manager and the container, which is using 0.4.7)
* Conversely make sure `shellcheck` is not in your $PATH to test the docker/podman mode.

## Deployment

No special considerations.

## Checklist

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container
- [x] These changes do not require documentation

### CODE REVIEW

File: devops/scripts/shellcheck.sh
Comment (cfm) [inline devops/scripts/shellcheck.sh]: Nit:  unnecessary `export`?  `$DOCKER_BIN` is only otherwise used in [`dev-shell` since #6216](https://github.com/freedomofpress/securedrop/pull/6216/files#diff-520aecbdbfb8d38a696ef31c20741145bdb40e99240ca7d5f7ddfbc207b79e9b), where it's defined based on the same `$USE_PODMAN` as here.  :-)  And your changes here remove the `bash -c` subprocessing.  All three modes work without this `export`.

**Review by cfm — DISMISSED** [review]
All three modes look good per the test plan!  Approving with one nit about which I'll ping you, @legoktm, before I merge.


## Native mode

* [x] Install `shellcheck` from your package manager and make sure it works in native mode. (this will likely spit out a bunch of errors because of the version mismatch between your package manager and the container, which is using 0.4.7)

```sh-session
cfm@ozymandias securedrop % git diff       
diff --git a/devops/scripts/rebase-ci.sh b/devops/scripts/rebase-ci.sh
index 4817bb76d..3ebd7cd12 100755
--- a/devops/scripts/rebase-ci.sh
+++ b/devops/scripts/rebase-ci.sh
@@ -1,6 +1,4 @@
 #!/bin/bash
-# shellcheck disable=SC2086
-#
 # To be utilized in CircleCI
 # Determine target branch from PR number and rebase against that
 
(.venv) cfm@ozymandias securedrop % shellcheck --version
ShellCheck - shell script analysis tool
version: 0.8.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net/
(.venv) cfm@ozymandias securedrop % make shellcheck
███ Linting shell scripts...

In ./devops/scripts/rebase-ci.sh line 23:
    target_branch="$(curl -s ${GITHUB_PR_URL} | python -c 'import sys, json; print(json.load(sys.stdin)["base"]["ref"])')"
                             ^--------------^ SC2086 (info): Double quote to prevent globbing and word splitting.
[...]
make: *** [shellcheck] Error 1
```


## Docker

* [x] Insert some buggy code in a shell script and run `make shellcheck` and confirm it still reports an error and exits with a non-zero status code

```sh-session
cfm@ozymandias securedrop % git diff       
diff --git a/devops/scripts/rebase-ci.sh b/devops/scripts/rebase-ci.sh
index 4817bb76d..3ebd7cd12 100755
--- a/devops/scripts/rebase-ci.sh
+++ b/devops/scripts/rebase-ci.sh
@@ -1,6 +1,4 @@
 #!/bin/bash
-# shellcheck disable=SC2086
-#
 # To be utilized in CircleCI
 # Determine target branch from PR number and rebase against that
 
cfm@ozymandias securedrop % make shellcheck
███ Linting shell scripts...

In ./devops/scripts/rebase-ci.sh line 23:
    target_branch="$(curl -s ${GITHUB_PR_URL} | python -c 'import sys, json; print(json.load(sys.stdin)["base"]["ref"])')"
                             ^-- SC2086: Double quote to prevent globbing and word splitting.

make: *** [shellcheck] Error 1
```

## Podman

* [x] Insert some buggy code in a shell script and run `make shellcheck` and confirm it still reports an error and exits with a non-zero status code

```sh-session
[user@disp2971 securedrop]$ git diff
diff --git a/devops/scripts/rebase-ci.sh b/devops/scripts/rebase-ci.sh
index 4817bb76d..3ebd7cd12 100755
--- a/devops/scripts/rebase-ci.sh
+++ b/devops/scripts/rebase-ci.sh
@@ -1,6 +1,4 @@
 #!/bin/bash
-# shellcheck disable=SC2086
-#
 # To be utilized in CircleCI
 # Determine target branch from PR number and rebase against that
 
[user@disp2971 securedrop]$ make shellcheck
███ Linting shell scripts...
/home/user/securedrop/devops/scripts/shellcheck.sh: line 20: docker: command not found
make: *** [Makefile:106: shellcheck] Error 127
[user@disp2971 securedrop]$ USE_PODMAN=1 make shellcheck
███ Linting shell scripts...

In ./devops/scripts/rebase-ci.sh line 23:
    target_branch="$(curl -s ${GITHUB_PR_URL} | python -c 'import sys, json; print(json.load(sys.stdin)["base"]["ref"])')"
                             ^-- SC2086: Double quote to prevent globbing and word splitting.

make: *** [Makefile:106: shellcheck] Error 1
```

File: devops/scripts/shellcheck.sh
Comment (legoktm) [reply devops/scripts/shellcheck.sh]: I think my first version of this patch just added podman support without the other refactoring, so it needed export since `run_native_or_in_docker` was being exported too. But now it's no longer necessary since we're not exporting that either! Thanks for catching.

**Review by cfm — APPROVED** [review]
Thanks, @legoktm!  I'll merge as soon as CI is green.

**Pull request merged**
